openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -124,3 +126,38 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+    def is_saml_user(self, user):
+        """
+        Check and return if the user has an associated SAML social auth record.
+
+        Args:
+            user: django.contrib.auth.models.User instance.
+        Returns:
+            True/False if the user has an associated SAML social auth record.
+        """
+        saml_user = False
+
+        if not user.is_authenticated:
+            return saml_user
+
+        user_social_auth = user.social_auth
+        social_auth_records = user_social_auth.select_related('user')
+
+        for social_auth in social_auth_records:
+            if social_auth.provider == 'tpa-saml':
+                saml_user = True
+
+        return saml_user
+
+    def allow_saml_logout_from_url(self):
```

Comment:
do you need the value to change in runtime ?? if the answer is not, you could define this as cached_property in order to avoid multiple calls to the configuration_helpers

Reply:
Good point, changed. @andrey-canon